### ISSUE TITLE
Bug in assigning a slice to a slice

### ISSUE THREAD

**Issue by awni** [arrayfire/arrayfire#2533]
The following code no longer works (the array `a` is not updated).

```
int main(int argc, char** argv) {

  af::info();
  af::array a(5, 10);
  auto b = af::randu(5, 10);
  a(af::span, af::seq(0, 5)) = b(af::span, af::seq(0, 5));
  af::print("", a);
  af::print("", b);
  return 0;
}
```
Output is:
```
ArrayFire v3.6.4 (CUDA, 64-bit Linux, build default)
Platform: CUDA Toolkit 9.2, Driver: 396.69
[0] Tesla M40, 11449 MB, CUDA Compute 5.2

[5 10 1 1]
    0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000
    0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000
    0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000
    0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000
    0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000     0.0000


[5 10 1 1]
    0.6010     0.5497     0.1583     0.3636     0.6755     0.8966     0.5143     0.0123     0.7917     0.2522
    0.0278     0.2864     0.3712     0.4165     0.6105     0.0536     0.3670     0.3988     0.1654     0.9644
    0.9806     0.3410     0.3543     0.5814     0.5232     0.5775     0.3336     0.9787     0.8657     0.4711
    0.2126     0.7509     0.6450     0.8962     0.5567     0.2908     0.0363     0.2308     0.3766     0.3637
    0.0655     0.4105     0.9675     0.3712     0.7896     0.9941     0.5349     0.6244     0.7331     0.9643
```

**Comment by awni** [comment]
Using an intermediate array works for now:

```
int main(int argc, char** argv) {

  af::info();
  af::array a(5, 10);
  auto b = af::randu(5, 10);
  auto c = b(af::span, af::seq(0, 5));
  a(af::span, af::seq(0, 5)) = c;
  af::print("", a);
  af::print("", b);
  return 0;
}
```

**Comment by 9prady9** [comment]
Closed via #2537
